src/System.Net.Sockets/src/System/Net/Sockets/SafeSocketHandle.Windows.cs

```diff
@@ -96,11 +96,9 @@ internal bool SkipCompletionPortOnSuccess
 
         internal static SafeSocketHandle CreateWSASocket(AddressFamily addressFamily, SocketType socketType, ProtocolType protocolType)
         {
-            SafeSocketHandle result = Interop.Winsock.WSASocketW(addressFamily, socketType, protocolType, IntPtr.Zero, 0, Interop.Winsock.SocketConstructorFlags.WSA_FLAG_OVERLAPPED | Interop.Winsock.SocketConstructorFlags.WSA_FLAG_NO_HANDLE_INHERIT);;
-            if (result.IsInvalid)
-            {
-                result.SetHandleAsInvalid();
-            }
+            SafeSocketHandle result = new SafeSocketHandle();
+            IntPtr handle = Interop.Winsock.WSASocketW(addressFamily, socketType, protocolType, IntPtr.Zero, 0, Interop.Winsock.SocketConstructorFlags.WSA_FLAG_OVERLAPPED | Interop.Winsock.SocketConstructorFlags.WSA_FLAG_NO_HANDLE_INHERIT);;
+            result.SetHandle(handle);
```

Comment:
This can be:
```C#
var result = new SafeSocketHandle(handle, ownsHandle: true);
```
There's no need to separately construct it first.

Reply:
I see; you're using your own `new SetHandle` to add additional checks.  In that case, please rename it to make it clear it's not the base's SetHandle (since that method isn't virtual and this adds additional functionality).  You can also make that method internal instead of private, and then follow through on my previous suggestion to get rid of this intermediate layer of functions.